VAMobile/documentation/docs/Engineering/FrontEnd/DevelopmentBestPractices.md

```diff
@@ -0,0 +1,130 @@
+
+Our team follows a set of best practices around tickets, branches, and PRs to ensure a smooth development process and high quality code. Here we'll describe the key points to keep in mind.
+
+## Tickets
+
+We use tickets to track all our engineering work. They precisely specify the work to be performed, allow stakeholders to track progress, and guide QA in assessing the resulting work.
+
+### Creating tickets
+
+Every ticket must contain at least the following information:
+
+- A detailed description of the work
+- Clearly specified Acceptance Criteria
+- Screenshots or links to mockups where applicable
+- All relevant labels ("front-end" for FE work, "Health" for Health and Benefits work, "Global" for global work, etc)
+- An epic relating to the ticket's larger context
+
+If possible, choose the relevant template for your ticket. For example, if you're creating a bug ticket, choose the Bug Report Template and include severity in the title, labels, and body.
+
+### Working on tickets
+
+All engineering work requires a ticket.
+
+Every ticket must satisfy the requirements in the "Creating tickets" section above. If information is missing, add it or contact the relevant stakeholders.
+
+Before an engineer can begin work on a ticket, it must contain an estimate. Bugs get an initial estimate of 1 to cover a short investigation. Other tickets are left to the engineers' discretion.
+
+Once everything is in place, complete the following tasks:
+
+- Assign yourself
+- Assign the appropriate QA engineer if not already assigned
+- Move the ticket to In Progress
+- Create a branch as described below in the Branches section
+
+Then you can begin work. Keep stakeholders up to date as you work by commenting on the ticket with any important details or questions. This will also preserve information so everyone can reference it in the future.
+
+On completion of your work, double check the following:
+
+- All items in Acceptance Criteria are implemented
+- Confirm screen screen reader functionality on both iOS and Android for accessibility
+- Unit and E2E tests are updated or extended to cover your changes
+- Relevant documentation is updated to reflect the changes in your PR
+
+Then create a Pull Request. See the Pull Requests section below for more information.
+
+## Branches
+
+Feature branches are temporary branches where we develop and test new features. We implement every code change in a feature branch before merging it into the default branch.
+
+:::info
+For features that must be rolled out gradually, or where precise feature activation timing is required, use a [remote config flag](https://firebase.google.com/docs/remote-config/).
+:::
+
+Our default branch is `develop`. Create your feature branch from there, named as follows:
+
+```
+type/ticketnumber-description-of-work
+```
+
+The ticket number lets us track back to the relevant ticket. The type and description show the branch's purpose at a glance. Here are some example branch names:
```

Comment:
Even though it's mostly self-explanatory, should we explicitly list out the types that we want people to name their branches under and their intended purposes, or should we allow people to use their own discretion? I know there are examples listed below, just not sure if that signifies these are types people _should_ be using.

Reply:
I think we should leave it intentionally vague. 

Narin and I recently had a similar discussion around [our PR template](https://github.com/department-of-veterans-affairs/va-mobile-library/blob/main/.github/pull_request_template.md?plain=1) listing `where Issue type = bug, feature, spike, CU (code upkeep), etc.` and agreed that the "etc." was good to allow some freedom. As long as there's some reasonable level of consistency so most tickets fall into similar buckets, it didn't seem worthwhile trying to be too prescriptive.

Reply:
If we start automating tasks based on the branch names, we'll probably need to get more strict about branch naming. I'm not too worried about it being perfect for now since there's no automation

Reply:
yes, please use a strict branch naming like we described above

Reply:
Added the following:
> Types are `feature`, `bug`, `chore`, and `hotfix`. Description is CamelCase with optional dashes to add further details.

Reply:
If this is supposed to be the totality of options, I disagree with the list. 

Some `types` we have used in the past that I feel do not fit those bins unless your definition of `chore` is extremely broad:
- spike
- cu (code upkeep)
- dependencies
- doc
- tests
- Detox (could be merged with tests?)
- storybook
- linting

Also, more broadly, the name `chore` seems very trivializing and dismissive unless the work is minor. There are tasks that are simultaneously not minor and not adding features/making functional changes.

An important part of a branch name is to be able to tell what it's about easily so having super vague/generic type categories is detrimental to classifying the work/being able to quickly identify what a branch is for.

Reply:
@timwright12 do you have a specific list of branch types you'd like us to use?

Reply:
let's start with: chore, feature, bug as they're industry standard

Reply:
@alexandec let's start with: `chore`, `feature`, and `bug` as they're industry standard